src/server/session.ts

```diff
@@ -677,6 +679,45 @@ module ts.server {
             }
         }
 
+        getDiagnosticsForProject(delay: number, fileName: string) {
+            let { configFileName, fileNameList } = this.getProjectInfo(fileName, true);
+            fileNameList = fileNameList.filter((value, index, array) => value.indexOf("lib.d.ts") < 0);
+
+            // Sort the file name list to make the recently touched files come first
+            let highPriorityFiles: string[] = [];
+            let mediumPriorityFiles: string[] = [];
+            let lowPriorityFiles: string[] = [];
+            let veryLowPriorityFiles: string[] = [];
+            let normalizedFilePath = ts.normalizePath(fileName);
+            let project = this.projectService.getProjectForFile(normalizedFilePath);
+            for (let oneFile of fileNameList) {
+                if (oneFile == normalizedFilePath)

```

Comment:
Thinking more into it I'm not sure how FileMap fits in here. As the only related operation is the one comparison. And the ``ts.normalizePath`` already includes normalize slashes. Let's discuss tomorrow.